File: training-data/ruby/perfect_numbers.rb
def aliquot_sum(n)
  return 0 if n == 1

  (2..Integer.sqrt(n)).select { |i| (n % i).zero? }.sum(1) { |i| i == n / i ? i : i + n / i }
end

def classify(n)
  case aliquot_sum(n) <=> n
  when 0 then :perfect
  when 1 then :abundant
  else :deficient
  end
end

[6, 12, 28, 13, 8128].each { |n| puts "#{n} #{classify(n)}" }

File: training-data/go/perfect_numbers.go
package main

import "fmt"

func aliquotSum(n int) int {
	if n == 1 {
		return 0
	}
	sum := 1
	for i := 2; i*i <= n; i++ {
		if n%i == 0 {
			sum += i
			if i != n/i {
				sum += n / i
			}
		}
	}
	return sum
}

func classify(n int) string {
	switch s := aliquotSum(n); {
	case s == n:
		return "perfect"
	case s > n:
		return "abundant"
	default:
		return "deficient"
	}
}

func main() {
	for _, n := range []int{6, 12, 28, 13, 8128} {
		fmt.Println(n, classify(n))
	}
}
